### ISSUE TITLE
Events not triggered after push

### ISSUE THREAD

**Issue by Kemcake** [Instagram/IGInterfaceDataTable#9]
This is probably due to the [swizzling](https://github.com/Instagram/IGInterfaceDataTable/issues/8) issue, but it seems that when you `push` a new controller with a `ig_datasource`, you completely lost events handling.
You can reproduce with the sample app by doing this: 

```
- (void)table:(WKInterfaceTable *)table didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
  [self pushControllerWithName:@"InterfaceController" context:nil];
  NSLog(@"Tapped row %@",indexPath);
}
```

**Comment by Kemcake** [comment]
A solution to deal with it can be to not implement `[self enableTableSelectCallbacks]` and call `didSelectRowAtIndexPath` manually like this:

```
-(void)table:(WKInterfaceTable *)table didSelectRowAtIndex:(NSInteger)rowIndex {
    NSIndexPath *indexPath = [table indexPathFromRowIndex:rowIndex];
    [self table:table didSelectRowAtIndexPath:indexPath];
}
```
However, still not a fix.

**Comment by rnystrom** [comment]
@Kemcake great find! Ya I'm not a fan of the swizzling at all, I'd like a better solution.

One option I explored was requiring you to implement `table:didSelectRowAtIndex:` and calling `super`, but that's not that great. I'm looking into another method/place that you can swizzle stuff.

**Comment by Kemcake** [comment]
Ok, found a solution, just need to implement swizzling in a `dispatch_once`, as seen on [NSHipster](http://nshipster.com/method-swizzling/#dispatch_once)

I'll do a pull request in few minutes after some tests :)

**Comment by rnystrom** [comment]
Closed in #10
